hathor/transaction/transaction.py

```diff
@@ -260,6 +263,33 @@ def to_json(self, decode_script: bool = False, include_metadata: bool = False) -
         json['tokens'] = [h.hex() for h in self.tokens]
         return json
 
+    def verify_basic(self, skip_block_weight_verification: bool = False) -> None:
+        if self.is_genesis:
+            # TODO do genesis validation?
+            return
+        self.verify_without_storage()
+        self.verify_parents_basic()
+        self.verify_weight()
+
+    def verify_parents_basic(self) -> None:
+        """Verify number and non-duplicity of parents."""
+        assert self.storage is not None
+
+        # check if parents are duplicated
+        parents_set = set(self.parents)
+        if len(self.parents) > len(parents_set):
+            raise DuplicatedParents('Tx has duplicated parents: {}', [tx_hash.hex() for tx_hash in self.parents])
+
+        if len(self.parents) != 2:
+            raise IncorrectParents(f'wrong number of parents (tx type): {len(self.parents)}, expecting 2')
+
+    def verify_weight(self) -> None:
```

Comment:
Same here: should it be part of the `verify_without_storage()`?

Reply:
For blocks it actually needs storage.

Reply:
Makes sense.

Reply:
I tried moving it and a bunch of tests broke. While I think the move makes sense and it might actually indicate something wrong on the tests, at this point I rather do this on a new PR.